TDP-3037 Recording feature (#226)
Bandwidth/bandwidth.github.io: voice/bxml/verbs/startRecording.md

{% method %}
## XML: `<StartRecording>`

> The StartRecording verb allows an entire section of a call to be recorded. All audio on both sides of the call will be recorded until either the call ends or the [`<StopRecording>`](stopRecording.md) verb is used.

Reviewer: ```suggestion
The StartRecording verb allows a segment of a call to be recorded while other verbs are executing. The audio from both sides of the call will be recorded until the [`<PauseRecording>`](pauseRecording.md) verb is executed, the [`<StopRecording>`](stopRecording.md) verb is executed, or the call ends.
```

| username                     | (optional) The username to send in the HTTP request to `recordCompleteUrl` or `recordingAvailableUrl`. If username and password are specified, then https is required.                                                 |
| password                     | (optional) The password to send in the HTTP request to `recordCompleteUrl` or `recordingAvailableUrl`. If username and password are specified, then https is required.                                                 |
| tag                          | (optional) A custom string that will be sent with this and all future callbacks unless overwritten by a future `tag` attribute or cleared.<br><br>May be cleared by setting `tag=""`<br><br>Max length 256 characters. |
| fileFormat                   | (optional) The format that the recording will be saved in - `mp3` or `wav`.  Default value is `wav`.                                                                                                                   |

> | multiChannel                 | (optional) A boolean value indicating whether or not the recording file should have each side of the call in its own audio channel. Default value is `false`.                                                          |

Reviewer: ```suggestion
| multiChannel                 | (optional) A boolean value indicating whether or not the recording file should separate each side of the call into its own audio channel. Default value is `false`. `true` results in two channels.                                                          |
```

If the `recordingAvailableUrl` attribute is specified, the [Recording Available](../callbacks/recordingAvailable.md) event is sent to the URL once the recording is available for download. No BXML is expected.

<aside class="alert general small"><p>NOTE: Only one &lt;StartRecording&gt; verb may be active at a time. If a second &lt;StartRecording&gt; verb is used without first using a &lt;StopRecording&gt; verb, the second &lt;StartRecording&gt; will be ignored.</p></aside>

### Callbacks Received

| Callbacks                                                         | Can reply with more BXML |
|:------------------------------------------------------------------|:-------------------------|
| [Recording Available](../callbacks/recordingAvailable.md)         | No                       |

{% common %}
#### Example: Recording of a call
This shows how to use Bandwidth XML to record a phone call.

```XML
<?xml version="1.0" encoding="UTF-8"?>
<Response>

>     <SpeakSentence voice="bridget">Transferring your call, please wait, this call is being recorded.</SpeakSentence>

Reviewer: ```suggestion
    <SpeakSentence voice="bridget">This call is being recorded. Please wait while we transfer you.</SpeakSentence>
```

    <StartRecording recordingAvailableUrl="https://myapp.com/noBXML"/>
    <Transfer>
        <PhoneNumber>+11234567892</PhoneNumber>
    </Transfer>
    <StopRecording/>

>     <SpeakSentence voice="bridget">Thanks for your call, have a nice day.</SpeakSentence>

Reviewer: ```suggestion
    <SpeakSentence voice="bridget">Thanks for your call. Have a nice day!</SpeakSentence>
```

| password                     | (optional) The password to send in the HTTP request to `recordingAvailableUrl`. If specified, the URL must be TLS-encrypted (i.e., `https`).                                                                           |
| tag                          | (optional) A custom string that will be sent with this and all future callbacks unless overwritten by a future `tag` attribute or cleared.<br><br>May be cleared by setting `tag=""`<br><br>Max length 256 characters. |
| fileFormat                   | (optional) The audio format that the recording will be saved as: `mp3` or `wav`.  Default value is `wav`.                                                                                                              |
| multiChannel                 | (optional) A boolean value indicating whether or not the recording file should separate each side of the call into its own audio channel. Default value is `false`. `true` results in two channels.                    |

> If the `recordingAvailableUrl` attribute is specified, then the [Recording Available](../callbacks/recordingAvailable.md) event is sent to the URL once the recording is available for download. BXML returned in response to this callback will be ignored.

Reviewer 2: or if there's an error

event is sent to the URL once the recording is available for download, indicating the `mediaUrl` and if there was any issue processing the recording.

BXML returned in response to this callback will be ignored.

<aside class="alert general small"><p>NOTE: Only one &lt;StartRecording&gt; verb may be active at a time. If a second &lt;StartRecording&gt; verb is used without first using a &lt;StopRecording&gt; verb, the second &lt;StartRecording&gt; will be ignored.</p></aside>

### Callbacks Received

| Callbacks                                                         | Can reply with more BXML |
|:------------------------------------------------------------------|:-------------------------|
| [Recording Available](../callbacks/recordingAvailable.md)         | No                       |

{% common %}
#### Example: Recording of a call
This shows how to use Bandwidth XML to record a phone call.

```XML
<?xml version="1.0" encoding="UTF-8"?>
<Response>
    <SpeakSentence voice="bridget">This call is being recorded. Please wait while we transfer you.</SpeakSentence>
    <StartRecording recordingAvailableUrl="https://myapp.com/noBXML"/>
    <Transfer>

>         <PhoneNumber>+11234567892</PhoneNumber>

Reviewer 2: it would be better to use +1555 numbers in all our examples

{% method %}
## XML: `<StartRecording>`
The StartRecording verb allows a segment of a call to be recorded while other verbs are executing.

> All audio on both sides of the call will be recorded until either the call ends or the [`<StopRecording>`](stopRecording.md) verb is used.

Reviewer: I think we should include `PauseRecording` in this description.
